### PR TITLE
Improve error message for missing wrong/named parameters

### PR DESCRIPTION
**Pull request by renegrob**
* If a named parameter is not specified you currently get a runtime error message: "Could not locate ordinal parameter [1], expecting one of []". With this PR you will get a compile time error message instead: "`myMethod` of Repository `MyRepo` is missing the named parameters [myParam], provided are [otherParam]. Ensure that the parameters are correctly annotated with `@Param`".
* The error messages of MethodNameParser did not include the Repository name which can make it hard to identify the source if you have many Repositories declaring the same method.

### CODE REVIEW

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java [hunk 1]
```diff
@@ -150,6 +153,19 @@ public void add(ClassCreator classCreator, FieldDescriptor entityClassFieldDescr
             try (MethodCreator methodCreator = classCreator.getMethodCreator(method.name(), methodReturnTypeDotName.toString(),
                     methodParameterTypesStr)) {
 
+                Set<String> usedNamedParameters = extractNamedParameters(queryString);
+                if (!usedNamedParameters.isEmpty()) {
+                    Set<String> missingParameters = new LinkedHashSet<>(usedNamedParameters);
+                    missingParameters.removeAll(namedParameterToIndex.keySet());
+                    if (!missingParameters.isEmpty()) {
+                        // better error message than "Could not locate ordinal parameter [1], expecting one of []"
```
Comment (geoand) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java:161]: I don't think we need the comment TBH - that comment is best left on the issue (as you have already done)

**Review by geoand — COMMENTED** [review]
Looks great!

I just added one tiny comment. When addressed, please take the PR out of draft so CI can run on it.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java (hunk 1 above)
Comment (renegrob) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java]: Yes, you're right. I removed it.

**Review by geoand — APPROVED** [review]
👌

**Pull request merged**